src/prototype_room_basebuilder.js

```diff
@@ -67,7 +67,7 @@ Room.prototype.destroyStructure = function(structure) {
     structuresMin = 1;
   }
 
-  if (structures.length > structuresMin) {
+  if (structures.length > structuresMin && Room.structureIsEmpty(structure)) {
```

Comment:
I think this breaks the case where the room layout changes and structures need to be placed somewhere else.

Hostile structures get removed after some time, owned structures can keep their resources and will be never replaced.

To address your case you could do `&& (structure.my || Room.structureIsEmpty(structure))` or something more sophisticated.